### PR TITLE
Avoid adding commit sha for NuGet CommandLine package which breaks nuget.exe self update

### PR DESCRIPTION
**Pull request by jainaashish**
Fixes - https://github.com/NuGet/Home/issues/7116 

@rrelyea

### CODE REVIEW

**Review by mishra14 — APPROVED** [review]

File: build/common.project.props [hunk 1]
```diff
@@ -178,8 +178,8 @@
       <_Parameter1>$(SemanticVersion).$(PreReleaseVersion)</_Parameter1>
     </AssemblyAttributes>
     <AssemblyAttributes Include="AssemblyInformationalVersion">
-      <_Parameter1 Condition="'$(BUILD_SOURCEVERSION)' == ''">$(SemanticVersion)$(PreReleaseInformationVersion)</_Parameter1>
-      <_Parameter1 Condition="'$(BUILD_SOURCEVERSION)' != ''">$(SemanticVersion)$(PreReleaseInformationVersion)+$(BUILD_SOURCEVERSION)</_Parameter1>
+      <_Parameter1>$(SemanticVersion)$(PreReleaseInformationVersion)</_Parameter1>
```
Comment (nkolev92) [inline build/common.project.props]: why change this? 
This wouldn't impact semver.

File: build/common.project.props [hunk 2]
```diff
@@ -199,8 +199,8 @@
     <!-- Assembly attributes for net core projects -->
     <AssemblyVersion>$(SemanticVersion).$(NetCoreAssemblyBuildNumber)</AssemblyVersion>
     <FileVersion>$(SemanticVersion).$(PreReleaseVersion)</FileVersion>
-    <InformationalVersion Condition="'$(BUILD_SOURCEVERSION)' == ''">$(SemanticVersion)$(PreReleaseInformationVersion)</InformationalVersion>
-    <InformationalVersion Condition="'$(BUILD_SOURCEVERSION)' != ''">$(SemanticVersion)$(PreReleaseInformationVersion)+$(BUILD_SOURCEVERSION)</InformationalVersion>
+    <InformationalVersion>$(SemanticVersion)$(PreReleaseInformationVersion)</InformationalVersion>
```
Comment (nkolev92) [inline build/common.project.props]: same here.

File: build/common.project.props (hunk 1 above)
Comment (jainaashish) [reply build/common.project.props]: trying different things, nothing has worked so far. I still cant figure out where it's adding that commitsha with package version.

If you know where it does, let me know!

File: build/common.project.props (hunk 2 above)
Comment (jainaashish) [reply build/common.project.props]: same as above

File: build/common.project.props [hunk 3]
```diff
@@ -138,7 +138,7 @@
 
   <PropertyGroup Condition=" '$(Version)' == '' ">
     <Version>$(SemanticVersion)$(PreReleaseInformationVersion)</Version>
-    <Version Condition="'$(BUILD_SOURCEVERSION)' != ''">$(SemanticVersion)$(PreReleaseInformationVersion)+$(BUILD_SOURCEVERSION)</Version>
+    <Version Condition="'$(BUILD_SOURCEVERSION)' != '' AND '$(IsCommandLinePackage)' != 'true'">$(SemanticVersion)$(PreReleaseInformationVersion)+$(BUILD_SOURCEVERSION)</Version>
```
Comment (nkolev92) [inline build/common.project.props:141]: It should be set here, but you have an evaluation ordering problem.

File: src/NuGet.Clients/NuGet.CommandLine/NuGet.CommandLine.csproj [hunk 4]
```diff
@@ -9,6 +9,7 @@
     <Shipping>true</Shipping>
     <PackageId>NuGet.CommandLine</PackageId>
     <GenerateDocumentationFile>false</GenerateDocumentationFile>
+    <IsCommandLinePackage>true</IsCommandLinePackage>
```
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.CommandLine/NuGet.CommandLine.csproj]: This needs to be set before the common props are imported

**Pull request merged**